code/modules/recycling/disposal/pipe.dm

```diff
@@ -93,11 +93,11 @@
 	var/eject_range = 5
 	var/turf/open/floor/floorturf
 
-	if(isfloorturf(T)) //intact floor, pop the tile
+	if(isfloorturf(T) && T.intact) //intact floor, pop the tile
 		floorturf = T
-		if(floorturf.floor_tile)
+		if(floorturf.floor_tile && floorturf.forced_plating == TRUE)
```

Comment:
```suggestion
		if(floorturf.floor_tile && floorturf.make_plating(floorturf.forced_plating))
```
make_plating returns null if it fails and true if it actually makes plating. You should take advantage of that here.

On another note, what's the purpose of forced_plating?

Reply:
its a check to ensure this is only being forced when its supposed to be forced, in testing, even if i did make_plating(force) floors were still poping their stack item even though they werent breaking, the var is an added check to ensure this isnt happening if its not actually breaking its tile.

Reply:
ah then checking the return value of make plating is a suitable replacement

Reply:
i couldnt figure out how to check the return value, no matter how you explained it about it returning null, i couldnt see *where* it said it returned null nor how to even check for that. and in testing that method i realized checking if a tile can make plating before spawning its stack item wasnt working quite as intended (it worked, but some tiles stopped spawning their stack item while some did) so im just going to force it at all times and have this be a problem for whoever makes the first floor that cant be popped from disposals pipes